File: 백준/Silver/10610. 30/30.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var s string
	var n, sum int
	fmt.Fscanf(reader, "%s\n", &s)

	if !strings.Contains(s, "0") {
		fmt.Println(-1)
	} else {
		sArr := strings.Split(s, "")
		sort.Slice(sArr, func(i, j int) bool {
			return sArr[i] > sArr[j]
		})
		for _, v := range sArr {
			n, _ = strconv.Atoi(v)
			sum += n
		}
		if sum%3 != 0 {
			fmt.Println(-1)
		} else {
			fmt.Println(strings.Join(sArr, ""))
		}
	}
}

File: 백준/Silver/10610. 30/30.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
n=list((input().strip('\n')))
n = sorted(n, reverse=True)
sum = 0
if '0' not in n:
    print(-1)
else:
    for i in n:
        sum += int(i)
    if sum % 3 != 0 : #3의 배수
        print(-1)
    else :
        print(''.join(n))